src/html/HTMLWriter.jl

```diff
@@ -921,18 +977,11 @@ function warning_script(src, ctx)
     if ctx.settings.warn_outdated
         return Tag(:script)[Symbol(OUTDATED_VERSION_ATTR), :src => relhref(src, ctx.warner_js)]()
     end
-    return Tag(Symbol("#RAW#"))("")
+    return empty_node()
 end
-
-function canonical_link_element(canonical_link, src)
-   @tags link
-   if canonical_link === nothing
-      return Tag(Symbol("#RAW#"))("")
-   else
-      canonical_link_stripped = rstrip(canonical_link, '/')
-      href = "$canonical_link_stripped/$src"
-      return link[:rel => "canonical", :href => href]
-   end
+ 
+function empty_node()
+    return Tag(Symbol("#RAW#"))("")
```

Comment:
This is a nice to have, but to make things a little neater here: do you think we could move this into the `DOM` module? We could have it be just a global `const VOID = Tag(Symbol("#RAW#"))("")` I think (as for the name, there is some precedent in HTML with "void elements").

We could also slightly modify the `@tags` macro to allow bringing `VOID` into the local context.

The `DOM` module has [some unit tests](https://github.com/JuliaDocs/Documenter.jl/blob/master/test/dom.jl) which could be used to make sure it all works as intended as well.